File: jasmine/lib/fizzbuzz.rb
def fizzbuzz(num)
  return 0 if num == '0'

  return nil if (num == nil) || (num.class == String && num.to_i == 0) || (num.class != String && num.class != Integer && num.class != Float)

  if num.class == String
    num = num.to_i
  end

  return 0 if num == 0

  return "fizzbuzz" if num % 3 == 0 && num % 5 == 0

  return "fizz" if num % 3 == 0

  return "buzz" if num % 5 == 0

  num

end

File: jasmine/src/fizzBuzz.js
// function FizzBuzz(num) {
//   if (num % 15 === 0) {
//     return 'fizzbuzz';
//   } else if (num % 3 === 0) {
//     return 'fizz';
//   } else if (num % 5 === 0) {
//     return 'buzz';
//   }
//   return num;
// };

function FizzBuzz() {}

FizzBuzz.prototype.play = function(number) {
  if (this._isDivisibleBy(number, 15)) {
    return 'FizzBuzz';
  } else if (this._isDivisibleBy(number, 3)) {
    return 'Fizz';
  } else if (this._isDivisibleBy(number, 5)) {
    return 'Buzz';
  }
  return number;
};

FizzBuzz.prototype._isDivisibleBy = function(number, divisor) {
  return number % divisor === 0;
};